### PR TITLE
Shared context and double spend state

### LINKED ISSUE #104: Per-origin and cross-origin implications on double spend state
Origins currently choose whether their challenges are per-origin or cross-origin. For cross-origin, double-spend prevention is only as good as the coordination between origins and the Issuer. Per-origin allows double-spend prevention to be isolated to a single origin; also prevents the cache of tokens being take up by some other origin.

### PR DESCRIPTION
**Pull request by chris-wood**
Closes #104.

### CODE REVIEW

File: draft-ietf-privacypass-architecture.md [hunk 1]
```diff
@@ -197,18 +197,20 @@ of ways in which the token may vary, including:
   are publicly verifiable, whereas others may not have this property.
 - Issuer identity. Tokens identify which issuers are trusted for a given
   issuance protocol.
-- Interactive or non-interactive. Tokens can either be interactive or not.
-  An interactive token is one which requires a freshly issued token based
-  on the challenge, whereas a non-interactive token can be issued proactively
-  and cached for future use.
+- Redemption context. Tokens can be redeemed for a given redemption context.
+  An empty redemption context effectively means tokens can be issued and redeemed
+  non-interactively, whereas a fresh and random redemption context means that
```
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-ietf-privacypass-architecture.md:202]: True, but I didn't want to enumerate all different options here. We do that in the auth-scheme draft already.

**Comment by chris-wood** [comment]
The section references seem to be breaking the build, so I'll fix that later before we land this.

File: draft-ietf-privacypass-architecture.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tfpauly) [reply draft-ietf-privacypass-architecture.md:202]: I agree that we shouldn't enumerate all cases, but I'd like to still clarify that there isn't a strict binary. Let me suggest some text.

File: draft-ietf-privacypass-architecture.md [hunk 2]
```diff
@@ -197,18 +197,20 @@ of ways in which the token may vary, including:
...
+  the redeemed token must be issued interactively. See {{Section 2.1.1 of HTTP-Authentication}}
```
Comment (chris-wood) [inline draft-ietf-privacypass-architecture.md:203]:
```suggestion
  the redeemed token must be issued interactively. See 2.1.1 of {{HTTP-Authentication}}
```

File: draft-ietf-privacypass-architecture.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-ietf-privacypass-architecture.md:202]: Looks good!

**Review by tfpauly — APPROVED** [review]

**Pull request merged**